gamemode/modules/drowning/sv_drowning.lua

```diff
@@ -0,0 +1,88 @@
+local firstSubmergedTime = {}
+local lastDrownTick = {}
+local config = GM.Config.Drowning
+
+local WATER_SUBMERGED = 3
+
+-- TODO: Add Gear specific rules for scuba gear and similar
+
+util.AddNetworkString( "RVR_Player_Enter_Water" )
+util.AddNetworkString( "RVR_Player_Leave_Water" )
+util.AddNetworkString( "RVR_Player_Take_Drown_Damage" )
+
+local function isDrowning( player )
+    return ( CurTime() - firstSubmergedTime[player:SteamID()] ) >= config.DROWNING_THRESHOLD
```

Comment:
is it possible for this to be called with firstSubmerged is nil

Reply:
The values are initialized in the tick hook if they don't exist